Add directory input option

nosmokingbandit opened the pull request:
The commit messages sum it up, but there are a few fixes here.

The loop iterating over form elements to find inputs wouldn't catch my 2nd input, but running the interator while counting down and not removing the element works.

The main change is the ability to pass a 'data-filemode' attribute to text inputs:
```
<input type="file" name="open_directory" id="open_directory" data-filemode="directory"/>
```

If 'data-filemode' is given and is "directory", the button will open a Choose Directory dialog. If 'data-filemode' is not given or isn't "directory" it will default to "file" and work as usual.

I also added a 'data-for' attribute to the generated buttons for inputs. This way additional css and javascript can select and modify those buttons by referring to their 'data-for' attribute. The value is the 'name' attribute of the input the button was generated for. Using the example above, the button will have the attribute 'data-for="open_directory"'.

This has been tested with python 2.7.12 on Windows 10, but I would expect these changes to be compatible with all platforms.


Also, I found in your docs that you use this as the data-filters parameter:
```
"[{'title': 'title for extension', 'extensions': 'space separated extensions'}, {'title': 'title for another extension', 'extensions': 'space separated extensions'}]"
```
However, since the data must be valid JSON if only works if given as
```
data-filter='[{"title": "title for extension", "extensions": "space separated extensions"}, {"title": "title for another extension", "extensions": "space separated extensions"}]'
```

nosmokingbandit committed:
Fix loop skipping inputs

I'm not entirely sure what was happening here (not having an inspector in QTWebkit is a pain, but looping over inputs would miss anything but the first input (it might catch them at intervals but I only have two text inputs on this project).

Making the iterator count down rather than up seems to have solved this. I don't know why, but it works.

nosmokingbandit committed:
Allows for input attribute data-filemode

User can set data-filemode='directory' in order to open a directory prompt instead of a file open prompt. It returns a str of the directory seleted. Falls back to 'file' mode if data-filemode not specified or incorrectly typed.

nosmokingbandit committed:
Add data-for attribute to buttons

Takes 'name' attribute from input and assigns it to button as 'data-for'. This allows for more creative css stlying and additional javascript interaction since it is easy to identify buttons.

nosmokingbandit committed:
Directory inputs no longer require data-filter to render correctly

It doesn't do anything anway, it just needs to have *some* kind of valid JSON.

amol-mandhane commented:
This looks like a backward incompatible change. Can you make it backward compatible?

Also, please squash the pull request into a single commit.

nosmokingbandit committed:
Allow for Directory input

Allows for input attribute data-filemode

User can set data-filemode='directory' in order to open a directory prompt instead of a file open prompt. It returns a str of the directory seleted. Falls back to 'file' mode if data-filemode not specified or incorrectly typed.

nosmokingbandit committed:
Merge remote-tracking branch 'origin/master'

nosmokingbandit committed:
Allow for Directory input

Allows for input attribute data-filemode

User can set data-filemode='directory' in order to open a directory prompt instead of a file open prompt. It returns a str of the directory seleted. Falls back to 'file' mode if data-filemode not specified or incorrectly typed.
Merge remote-tracking branch 'origin/master'

nosmokingbandit committed:
Merge remote-tracking branch 'origin/master'
